File: src/app/models/message/message-response.ts
import {AnswerResponse} from "../answer/answer-response";

export interface MessageResponse {
  id: number;
  messageText: string;
  subject: string;
  authorId:number;
  recordUrl?: string;
  imageUrl?: string;
  videoUrl?: string;
  pdfUrl?: string;
  createdAt: Date;
  updatedAt: Date;
  authorFullName: string;
 // answers: AnswerResponse[]; // New property for answers
    editMode?: boolean; // Add this line if it doesn't already exist


    showAnswers: boolean;
    answers?: AnswerResponse[];
    forumId: number;
    lastMessageDate: Date;

}

File: src/main/java/com/pfe/elearning/forum/Message/MessageResponse.java
package com.pfe.elearning.forum.Message;

import com.pfe.elearning.forum.answer.AnswerResponse;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MessageResponse {
    private Integer id;
    private String subject;
    private String messageText;
    private String recordUrl;
    private String imageUrl;
    private String videoUrl;
    private String pdfUrl;
  //  private MessageResponse message; // Ensure this field is present
    private String lessonTitle;
    //private String message;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Integer authorId;
    private String authorFullName; // Username of the discussion creator
    private Integer forumId;
    private List<AnswerResponse> answers; // Add list of answers

}